# HG changeset patch
# Parent 652249d09cb2d7529f17e78150cc4484f66eb976
# User Ben Kelly <ben@wanderview.com>
Fix link issue with FetchBody<derived>() on some platforms.


diff --git a/dom/fetch/Request.cpp b/dom/fetch/Request.cpp
--- a/dom/fetch/Request.cpp
+++ b/dom/fetch/Request.cpp
@@ -25,18 +25,17 @@ NS_IMPL_CYCLE_COLLECTING_RELEASE(Request
 NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(Request, mOwner, mHeaders)
 
 NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(Request)
   NS_WRAPPERCACHE_INTERFACE_MAP_ENTRY
   NS_INTERFACE_MAP_ENTRY(nsISupports)
 NS_INTERFACE_MAP_END
 
 Request::Request(nsIGlobalObject* aOwner, InternalRequest* aRequest)
-  : FetchBody<Request>()
-  , mOwner(aOwner)
+  : mOwner(aOwner)
   , mRequest(aRequest)
 {
 }
 
 Request::~Request()
 {
 }
 
diff --git a/dom/fetch/Response.cpp b/dom/fetch/Response.cpp
--- a/dom/fetch/Response.cpp
+++ b/dom/fetch/Response.cpp
@@ -26,18 +26,17 @@ NS_IMPL_CYCLE_COLLECTING_RELEASE(Respons
 NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(Response, mOwner, mHeaders)
 
 NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(Response)
   NS_WRAPPERCACHE_INTERFACE_MAP_ENTRY
   NS_INTERFACE_MAP_ENTRY(nsISupports)
 NS_INTERFACE_MAP_END
 
 Response::Response(nsIGlobalObject* aGlobal, InternalResponse* aInternalResponse)
-  : FetchBody<Response>()
-  , mOwner(aGlobal)
+  : mOwner(aGlobal)
   , mInternalResponse(aInternalResponse)
 {
 }
 
 Response::~Response()
 {
 }
 
